server/api/src/test/java/org/hawkular/apm/server/api/utils/ProducerInfoUtilTest.java

```diff
@@ -0,0 +1,234 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.server.api.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.hawkular.apm.api.model.Constants;
+import org.hawkular.apm.api.model.events.ProducerInfo;
+import org.hawkular.apm.server.api.model.zipkin.Annotation;
+import org.hawkular.apm.server.api.model.zipkin.BinaryAnnotation;
+import org.hawkular.apm.server.api.model.zipkin.Span;
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.SpanCache;
+import org.junit.Test;
+
+/**
+ * @author gbrown
+ */
+public class ProducerInfoUtilTest {
+
+    @Test
+    public void testGetProducerInfoNoClient() {
+        SpanCache spanCache = new TestSpanCache();
+
+        Span serverSpan = new Span();
+        serverSpan.setId("1");
+        serverSpan.setParentId("1");
+
+        List<Span> spans = new ArrayList<Span>();
+        spans.add(serverSpan);
+
+        try {
+            spanCache.store(null, spans, SpanUniqueIdGenerator::toUnique);
+        } catch (CacheException e) {
+            fail("Failed to store span");
+        }
+
+        assertNull(ProducerInfoUtil.getProducerInfo(null, serverSpan, spanCache));
+    }
+
+    @Test
+    public void testGetProducerInfoJustClient() {
+        SpanCache spanCache = new TestSpanCache();
+
+        Span serverSpan = new Span();
+        serverSpan.setId("1");
+        serverSpan.setParentId("1");
+        serverSpan.setAnnotations(serverAnnotations());
+
+        Span clientSpan = new Span();
+        clientSpan.setId("1");
+        clientSpan.setParentId("1");
+        BinaryAnnotation ba = new BinaryAnnotation();
+        ba.setKey("http.url");
+        ba.setValue("http://myhost:8080/myuri");
+        clientSpan.setBinaryAnnotations(Arrays.asList(ba));
+        clientSpan.setAnnotations(clientAnnotations());
+
+        try {
+            spanCache.store(null, Arrays.asList(serverSpan, clientSpan),
+                    SpanUniqueIdGenerator::toUnique);
+        } catch (CacheException e) {
+            fail("Failed to store span");
+        }
+
+        ProducerInfo pi = ProducerInfoUtil.getProducerInfo(null, serverSpan, spanCache);
+
+        assertNotNull(pi);
+        assertEquals(Constants.URI_CLIENT_PREFIX + "/myuri", pi.getSourceUri());
```

Comment:
No, if you take a service A and service B, the producer in A called 'urlB' - the sourceURI is the URI used to call service A - it is used as a way to identify which service is calling service B's URL.